lib/vector-core/src/event/test/serialization.rs

```diff
@@ -46,44 +47,80 @@ fn event_can_go_from_raw_prost_to_encodable_and_vice_versa() {
         "key3" => "value3",
     };
     let event: Event = LogEvent::from_parts(event_fields, EventMetadata::default()).into();
+    let events = EventArray::from(event);
 
     // First test: raw Prost encode -> `Encodable::decode`.
-    let first_event = event.clone();
-
     let mut first_encode_buf = BytesMut::with_capacity(4096);
-    proto::EventWrapper::from(first_event)
+    proto::EventArray::from(events.clone())
         .encode(&mut first_encode_buf)
-        .expect("event should not fail to encode");
+        .expect("events should not fail to encode");
 
     let first_decode_buf = first_encode_buf.freeze();
-    let first_decoded_event = decode_value::<Event, _>(first_decode_buf);
+    let first_decoded = EventArray::decode(EventArray::get_metadata(), first_decode_buf)
+        .expect("events should not fail to decode");
 
-    assert_eq!(event, first_decoded_event);
+    assert_eq!(events, first_decoded);
 
     // Second test: `Encodable::encode` -> raw Prost decode.
-    let second_event = event.clone();
-
     let mut second_encode_buf = BytesMut::with_capacity(4096);
-    encode_value(second_event, &mut second_encode_buf);
+    events
+        .clone()
+        .encode(&mut second_encode_buf)
+        .expect("events should not fail to encode");
 
     let second_decode_buf = second_encode_buf.freeze();
-    let second_decoded_event: Event = proto::EventWrapper::decode(second_decode_buf)
+    let second_decoded: EventArray = proto::EventArray::decode(second_decode_buf)
         .map(Into::into)
-        .expect("event should not fail to decode");
+        .expect("events should not fail to decode");
+
+    assert_eq!(events, second_decoded);
+}
+
+#[test]
+fn event_can_go_from_raw_prost_to_eventarray_encodable() {
+    // This is another test that ensures that we can encode via a raw Prost encode call and decode
+    // via `EventArray`'s `Encodable` implementation, specifically for a single `Event`.  This is
+    // the invariant we must provide to ensure that older disk buffer v1 files are still readable
+    // when `EventArray` is introduced.
+    //
+    // REVIEWERS: Be aware, if this is being removed or changed, the only acceptable context is
+    // LevelDB-based disk buffers being removed, or some other extenuating circumstance that must be
+    // explained.
```

Comment:
Procedural signoff: we're changing this because we need to be able to actually write an EventArray to a disk buffer. Since that's the new type being used, we're changing our check to ensure that the metadata used for encoding EventArray can properly be roundtripped.

As the comments indicate, the `Encodable` impl for `EventArray` falls back to decoding as `Event` which allows for interoperation with disk buffers created by versions of Vector where `EventArray` did not yet exist and was not yet used.